### ISSUE TITLE
HowTo: Start Granatum on Cluster

### ISSUE THREAD

**Issue by subwaystation** [lanagarmire/Granatum#2]
Hi,
I would like to start the Granatum server on our cluster. However, I do not have a Virtualbox installation at hand, just R.
Is there a way How I could install Granatum?
Thanks!
Best,
Simon

**Comment by w9** [comment]
Hi @subwaystation, I'm Xun, the lead developer of Granatum.

Technically it's possible to install Granatum locally without VirtualBox, but that is not recommended. In doing so, you would have to install and configure all the dependencies (R-packages, Linux binaries and dynamic libraries) yourself. It would have been much easier to just install VirtualBox and import the ova file.

It's easier than you think! Takes you one minute maximum. Check out this video on how to install VirtualBox and import OVA files: https://www.youtube.com/watch?v=Wz_NUue_OOo

OVA file: https://drive.google.com/file/d/0BzaGo-kaBuGLZ2EtUEZPUDNpSW8/view?usp=sharing

Let us know how it works for you.

**Comment by subwaystation** [comment]
Hi @w9,
thanks for the link. I already tried that one out and it works like a charm on my laptop.
However, VirtualBox is not available on the cluster and X-Port forwarding is not possible.
That is why I am asking. 
Maybe containerization (preferably Singularity) would be a more feasible way to bring the software to the cluster?
Thanks!

PS: I understand that installing the whole package by hand would be of too much hassle.

**Comment by w9** [comment]
Hi @subwaystation, are you familiar with Docker? It's technically possible to convert a VirtualBox into a Docker image, using something like [this](https://github.com/djui/docker-vbox-img).

Next week we will try to see if we can provide that for download. Meanwhile you are welcome to give this possibility a try.

**Comment by subwaystation** [comment]
Hi  @w9,
I was not aware that Docker supports this, thanks!
For security reasons I am only allowed to run Singularity but not Docker on the cluster, but I think I should be able to port the Docker to Singularity (http://singularity.lbl.gov/docs-docker). As soon as I have more news, I will update you.
Best.

**Comment by apeltzer** [comment]
Could you at least provide some installation documentation in terms of what kind of dependencies are used to set up the pipeline? That would be great and user friendly ;-) 

While providing a VirtualBox image is a good thing indeed, this does not scale well when users would like to use cluster or HPC environments. In that case, you'd need to either provide a working Docker Image or a Singularity container with granatum installed for users. 

Is there some kind of a batch mode where you can furthermore upload multiple files via a CLI to then analyze the data simultaneously or is Granatum more like an interactive work bench?

**Comment by w9** [comment]
@apeltzer For now Granatum focuses mainly on being a interactive platform. The Granatum 2 we are building is going to be much more ambitious and may support CLI batch analysis. Meanwhile there are some existing programmable tools like [Seurat](http://satijalab.org/seurat/) which performs some steps in Granatum.

For the installation, there are actually just four steps:

1) Install R
2) Install all the R-packages (as listed [here](https://github.com/lanagarmire/Granatum/blob/master/load_dependencies.R))
3) Download the Granatum source code [here](https://github.com/lanagarmire/Granatum)
4) Run `Rscript shiny.R`

However, the tricky part is that you most likely need the exact same versions of R and all the packages as are the ones in that VirtualBox OVA file. I'm not aware of a simple and reliable way to do this.

**Comment by lanagarmire** [comment]
Simon,

Thank you very much for your feedback and suggestions. We really appreciate
the interactions with Granatum users, and hope to make it a widely
applicable for everyone, from developers to end users.
Please feel free to email us any time with suggestions and feedbacks.
Best
Lana

**Comment by lanagarmire** [comment]
Agreed. Although installing source code is least recommended, we will get
the documentation up online soon. Thomas, please put this as a to-do task.

Lana

On Nov 9, 2017 11:20 PM, "Alexander Peltzer" <notifications@github.com>
wrote:

Could you at least provide some installation documentation in terms of what
kind of dependencies are used to set up the pipeline? That would be great
and user friendly ;-)

While providing a VirtualBox image is a good thing indeed, this does not
scale well when users would like to use cluster or HPC environments. In
that case, you'd need to either provide a working Docker Image or a
Singularity container with granatum installed for users.

Is there some kind of a batch mode where you can furthermore upload
multiple files via a CLI to then analyze the data simultaneously or is
Granatum more like an interactive work bench?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/lanagarmire/Granatum/issues/2#issuecomment-343420679>,
or mute the thread

**Comment by apeltzer** [comment]
Thanks! I know this is somewhat taking some time, but there are others probably also coming down the same road with us at some point.
